### PR TITLE
Define data classes and data loader

### PR DESCRIPTION
**Pull request by lolainta**
@yungyuc

### CODE REVIEW

**Comment by yungyuc** [comment]
It is counter-productive to review code without annotation.  I generally do not review a diff that is larger than 500 lines of code, because it is too hard to get it done in a day.

It is good that you filed the PR and let me know, so that I had the opportunity why it should not work in this way.  It seems to me that you have not had much experience to write production code collaboratively.  Do you think you want to gain some experience?

File: include/Annotation.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const std::string &get_visibility_token() const;
+  const std::vector<std::string> &get_attribute_tokens() const;
+  const Translation &get_translation() const;
+  const Rotation &get_rotation() const;
+  const std::array<double, 3> &get_size() const;
+  const std::string &get_prev() const;
+  const std::string &get_next() const;
+  Annotation() = delete;
+  Annotation(const json &);
+  Annotation(std::string, std::string, std::string, std::string,
```
Comment (yungyuc) [inline include/Annotation.hpp:36]: Passing `std::string` to a constructor by value induces unnecessary copy for initialization member data.

File: include/Annotation.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const std::string &get_sample_token() const;
+  const std::string &get_instance_token() const;
+  const std::string &get_visibility_token() const;
+  const std::vector<std::string> &get_attribute_tokens() const;
+  const Translation &get_translation() const;
+  const Rotation &get_rotation() const;
+  const std::array<double, 3> &get_size() const;
+  const std::string &get_prev() const;
+  const std::string &get_next() const;
+  Annotation() = delete;
```
Comment (yungyuc) [inline include/Annotation.hpp:34]: By rule of five, if there is a custom default constructor, you should supply other constructors and assignment operators.

**Comment by lolainta** [comment]
> It is counter-productive to review code without annotation. I generally do not review a diff that is larger than 500 lines of code, because it is too hard to get it done in a day.


Thanks for advisement.

I had considered splitting it into few PR originally. However, since all of those definitions of classes are similar and may cause more than 10 PRs or more. Thus, I only split them in commit level and merge it with one PR only. I will take this into consideration next time.

> It is good that you filed the PR and let me know, so that I had the opportunity to make it clear why it should not work in this way. It seems to me that you have not had much experience to write production code collaboratively. Do you think you want to gain some experience?

Yes, I can looking for more development experience, but I hope I can start it one month later. There're many deadline I need to deal with this month. 🥵

File: include/Annotation.hpp (hunk 1 above)
Comment (lolainta) [reply include/Annotation.hpp:36]: Thanks, I'll fix it and avoid unnecessary copy it in the next PR.

File: include/Annotation.hpp (hunk 2 above)
Comment (lolainta) [reply include/Annotation.hpp:34]: Yes, I had met some issue and added the `operator = ` for every class in #6. 

However, this is my first time hearing the rule of five, I'll look into it.

File: include/Annotation.hpp (hunk 1 above)
Comment (lolainta) [reply include/Annotation.hpp:36]: I've merged it in #7

File: include/Annotation.hpp (hunk 2 above)
Comment (yungyuc) [reply include/Annotation.hpp:34]: I will try to make my lecture more attractive next semester to help students to capture that when I show it to you 🙂

**Pull request merged**